### PR TITLE
feat: Allow configuring NiFi proxy host behaviour

### PR DESCRIPTION
**Pull request by labrenbe**
# Description

Allow adding proxy hosts that should be allowed by NiFi or turning the host header check off entirely.

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes

```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] CRD documentation for all fields, following the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs/style-guide).
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
- [ ] Changes need to be "offline" compatible
```

```[tasklist]
# Reviewer
- [x] Code contains useful comments
- [x] Code contains useful logging statements
- [x] (Integration-)Test cases added
- [x] Documentation added or updated. Follows the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs/style-guide).
- [x] Changelog updated
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
- [ ] [Roadmap](https://github.com/orgs/stackabletech/projects/25/views/1) has been updated
```

### CODE REVIEW

**Comment by maltesander** [comment]
Could we add a test for this (e.g. set a proxy header in the smoke test and check if its set in the pod / configmap / settings- see https://github.com/stackabletech/trino-operator/blob/main/tests/templates/kuttl/resources/20-assert.yaml)?

**Comment by labrenbe** [comment]
> Could we add a test for this (e.g. set a proxy header in the smoke test and check if its set in the pod / configmap / settings- see https://github.com/stackabletech/trino-operator/blob/main/tests/templates/kuttl/resources/20-assert.yaml)?

I've added an extra TestAssert step to the smoke test to not spam the test logs during `30-assert.`

File: rust/operator-binary/src/controller.rs [hunk 1]
```diff
@@ -1364,16 +1364,25 @@ async fn get_proxy_hosts(
     nifi: &NifiCluster,
     nifi_service: &Service,
 ) -> Result<String> {
+    let host_header_check = nifi.spec.cluster_config.host_header_check.clone();
+
+    if host_header_check.allow_all {
+        tracing::info!("spec.clusterConfig.hostHeaderCheck.allowAll is set to true. All proxy hosts will be allowed.");
```
Comment (maltesander) [inline rust/operator-binary/src/controller.rs:1370]: We could also trace that `additionalAllowedHosts` will be ignored.

**Review by maltesander — COMMENTED** [review]
LGTM overall. I would say (and with regard to the many support cases) to add a small documentation entry (explaining the whole `*` allow all behavior since this is patched by us and ready to be picked up by SEO)?
Edit: Changelog missing!

File: rust/operator-binary/src/controller.rs (hunk 1 above)
Comment (labrenbe) [reply rust/operator-binary/src/controller.rs:1370]: I've added an additional trace only when `additionalAllowedHosts` is not empty and will be ignored. My assumption is that this is then probably not behavior intended by the user.

File: docs/modules/nifi/pages/usage_guide/security.adoc [hunk 2]
```diff
@@ -167,3 +167,19 @@ sensitiveProperties:
   keySecret: nifi-sensitive-property-key
   algorithm: nifiArgon2AesGcm256
 ----
+
+[#host-header-check]
+== Host Header Check
+NiFi checks the Host header of incoming requests and rejects them if they are passing through a proxy that is not on an allow-list configured in the `nifi.web.proxy.host` property.
+
+A https://github.com/stackabletech/docker-images/pull/694[patch] applied during the build of the SDP container image for NiFi allows turning off this check by adding `nifi.web.proxy.host=*` to the properties. The Host header check for NiFi clusters created by the operator is turned off by default but can be turned in the NiFi configuration. In this case the list of allowed hosts will default to Kubernetes Services used by Nifi and can be extended with custom entries.
```
Comment (maltesander) [inline docs/modules/nifi/pages/usage_guide/security.adoc:175]:
```suggestion
A https://github.com/stackabletech/docker-images/pull/694[patch] applied during the build of the SDP container image for NiFi allows turning off this check by adding `nifi.web.proxy.host=*` to the properties. The Host header check for NiFi clusters created by the operator is disabled by default but can enabled in the NiFi configuration. In this case the list of allowed hosts will default to Kubernetes Services used by NiFi and can be extended with custom entries.
```

File: docs/modules/nifi/pages/usage_guide/security.adoc [hunk 3]
```diff
@@ -167,3 +167,19 @@ sensitiveProperties:
   keySecret: nifi-sensitive-property-key
   algorithm: nifiArgon2AesGcm256
 ----
+
+[#host-header-check]
+== Host Header Check
+NiFi checks the Host header of incoming requests and rejects them if they are passing through a proxy that is not on an allow-list configured in the `nifi.web.proxy.host` property.
```
Comment (maltesander) [inline docs/modules/nifi/pages/usage_guide/security.adoc:173]:
```suggestion
NiFi checks the host header of incoming requests and rejects them if they are passing through a proxy that is not on an allow-list configured in the `nifi.web.proxy.host` property.
```

**Review by maltesander — CHANGES_REQUESTED** [review]

**Review by maltesander — APPROVED** [review]
Nice thanks, LGTM!

**Pull request merged**